src/ui/components/IdentitiesContent/Item/index.tsx

```diff
@@ -48,6 +62,18 @@ export const IdentityItem = ({
     onSelectIdentity(commitment);
   }, [commitment, onSelectIdentity]);
 
+  const handleSelectedToConnect = useCallback(async () => {
+    if (!host) {
+      throw new Error("Please set host in order to continue this action.");
+    }
+    if (handleSelection) {
+      handleSelection(commitment);
+    }
+
+    await dispatch(setSelectedToConnect({ commitment, host }));
+    await dispatch(setNotReadyToConnect(false));
+  }, [commitment, dispatch, setSelect]);
```

Comment:
Please, check all the deps for `useCallback` and `useEffect` in this PR. Most deps are not there or used with different names.